### DISCUSSION TITLE
How to use "additional action" from permissions in a plugin?

### DISCUSSION THREAD

**Discussion by SaschaSchwarzK** [netbox-community/netbox#16300] [Q&A]
Hi, does anybody know what would be the "official" way to have custom permissions in a plugin? The problem I have is that all permission will be generated based on the models. But what if you have a need to create more than view, add, change, delete permissions?
There is the "Additional Actions"  in the permission and the documentation states that is exactly for this purpose. But I was not able to find an example how to create and use those in your plugin.
And of course I also would like to use something that is supposed to stay, not planned to be depreciated.

**Comment by candlerb** [comment]
I note that the plugin https://github.com/netbox-community/netbox-floorplan-plugin has custom permissions, you could look at how it uses them.

There was a diff here: https://github.com/netbox-community/netbox-floorplan-plugin/commit/b332f7edbe239d06c20cc7cf976bd2ff4b0996d5

**Comment by wiedi** [comment]
I was wondering the same thing recently and figured out the following:

The "Additional Actions" can be a list of comma separated values, of which each will be combined with the selected "Object Type" (app + model) so that you have `<app>.<additional_action>_<model>`.
For example if you create a Permission for "DCIM -> Device" with the "Additional Action" set to "refresh" your views would need to check for `"dcim.refresh_device"`.
